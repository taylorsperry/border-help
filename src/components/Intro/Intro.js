import React from 'react'
import Paragraph from '../Paragraph/Paragraph'
import shortid from 'shortid'
import PropTypes from 'prop-types'

export const Intro = ({data}) => {
  
  const displayIntro = data.map(paragraph => 
    <Paragraph key={shortid.generate()} para={paragraph.para}></Paragraph>
    )
  return(
    <div className='category'>
      <h2 className='cat-name'>Introduction</h2>
      {displayIntro}
      <p className='footer'>Data collected from the ACLU</p>
    </div>
  )
}

Intro.propTypes = {
  data: PropTypes.array
}

export default Intro;