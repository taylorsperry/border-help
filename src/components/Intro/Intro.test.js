import React from 'react'
import { Intro } from './Intro'
import { shallow } from 'enzyme'

describe('Intro', () => {
  let wrapper
  it('should match the snapshot', () => {
    let mockProps = {
      data: ['one', 'two']
    }
    wrapper = shallow(<Intro {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })
})