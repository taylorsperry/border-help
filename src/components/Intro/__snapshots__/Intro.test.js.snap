// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro should match the snapshot 1`] = `
<div
  className="category"
>
  <h2
    className="cat-name"
  >
    Introduction
  </h2>
  <Paragraph
    key="a66h8v2PN"
  />
  <Paragraph
    key="FBLlYWl3SF"
  />
  <p
    className="footer"
  >
    Data collected from the ACLU
  </p>
</div>
`;
